### PR TITLE
tests: Check all GAPIC clients can be created successfully

### PR DESCRIPTION
**Pull request by jskeet**
This will prevent failures of HTTP configuration for REGAPIC clients.

### CODE REVIEW

**Comment by jskeet** [comment]
Note: this initially deliberately breaks Compute, so we can see what a failure will look like.

File: tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        {
+            throw new UserErrorException($"No such API: {id}");
+        }
+        if (api.Generator != GeneratorType.Micro)
+        {
+            Console.WriteLine($"{id} is not a GAPIC-generated API; skipping client creation test.");
+        }
+
+        var assembly = PublishAndLoadAssembly(id);
+        if (assembly is null)
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs:48]: nit: Maybe a note on when this could happen?

**Comment by jskeet** [comment]
Humbug - failed for the wrong reason. Amending...

**Comment by jskeet** [comment]
Okay, that's more like it. The deliberately-introduced failure mode reported this:

> Configuration error: Package 'Google.Cloud.Compute.V1' is broken: creation of client 'AcceleratorTypesClient' with transport 'Rest' failed: System.ArgumentException: Field no-such-name not found in message google.cloud.compute.v1.AggregatedListAcceleratorTypesRequest

Now reverting the Compute change...

File: tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            .Where(t => t.IsClass)
+            .Where(t => t.Name.EndsWith("Client"))
+            .Where(t => assembly.GetType(t.FullName + "Builder") is Type builderType
+                        && builderType.BaseType.Name == "ClientBuilderBase`1")
+            .ToList();
+
+        if (clientTypes.Count == 0)
+        {
+            throw new UserErrorException($"API {id} has no client types.");
+        }
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs:78]: Can we just do nothing here? Thinking of the log proto packages, etc., so we don't have to filter them out in bash possibly?

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs (hunk 2 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs:78]: They'll be filtered out earlier, by not using the GAPIC generator. (I believe, anyway. I'm running build-everything run now.)

File: tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs (hunk 1 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs:48]: Ooh, this was left over from a previous iteration. Will remove.

File: tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs (hunk 2 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs:78]: Ha... that was the *plan*, anyway. But:

```text
Google.Apps.Script.Type is not a GAPIC-generated API; skipping client creation test.
Configuration error: API Google.Apps.Script.Type has no client types.
```

Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs:78]: (The joys of a missing return statement.)

**Pull request merged**